src/agent/rustjail/src/container.rs

```diff
@@ -1448,18 +1475,40 @@ impl LinuxContainer {
 
         let linux = spec.linux.as_ref().unwrap();
 
-        let cpath = if linux.cgroups_path.is_empty() {
-            format!("/{}", id.as_str())
+        // determine which cgroup driver to take and then assign to config.use_systemd_cgroup
+        // systemd: "[slice]:[prefix]:[name]"
+        // fs: "/path_a/path_b"
+        let cpath = if SYSTEMD_CGROUP_PATH_FORMAT.is_match(linux.cgroups_path.as_str()) {
```

Comment:
This should be determined when the agent starts (and save as a sandbox property) instead of when creating each container.

Reply:
There is currently no explicit flag to indicate whether the agent should use systemd cgroup, but instead decides based on the `linux.cgroups_path` in the container's `config.json` when creating the container. Should I modify the logic so that the sandbox decides whether the entire sandbox uses systemd when the first container is created?

Reply:
In the current implementation, it is required that the guest have to follow the same cgroup manager choice as the host. But what if the host asks for systemd cgroup but the guest only supports cgroup fs, or vice versa? So correct me if I'm wrong, but I think a better choise would be

Reply:
In the current implementation, it is required that the guest have to follow the same cgroup manager choice as the host. But what if the host asks for systemd cgroup but the guest only supports cgroup fs? IIUC, systemd cgroup manager is actually a guest capability that may or may not exist (depending on the guest rootfs image setup). So a better choice might be to determine it when creating the guest VM, let the runtime know it and let the runtime decide which cgroup manager to use. So that we can still run kata-containers w/o guest systemd in a systemd host environment.

I admit that it is too much for the current PR so it is fine to merge it as it, adding a requirement that anyone who wants to run systemd cgroup manager must enable it both on the host and in the guest. And we can improve it incrementally in the future.